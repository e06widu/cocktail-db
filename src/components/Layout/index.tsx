import React, { ReactNode } from 'react'
import Header from '../Header'

type LayoutProps = {
    children: ReactNode;
  };

  const Layout: React.FC<LayoutProps> = ({ children }) => {
    return (
        <div>
            <Header />
            <main className="main">{children}</main>
        </div>
    )
}

export default Layout
