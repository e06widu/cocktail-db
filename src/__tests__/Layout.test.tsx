import { render } from '@testing-library/react';
import Layout from '../components/Layout';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter as Router } from 'react-router-dom';

const queryClient = new QueryClient();

describe('Layout', () => {
    test('renders Header component', () => {
        const { container } = render(
          <QueryClientProvider client={queryClient}>
            <Router>
              <Layout>
                <div data-testid="child">Child Component</div>
              </Layout>
            </Router>
          </QueryClientProvider>
        );
    
        // Verify the rendered component matches the snapshot
        expect(container.firstChild).toMatchSnapshot();
      });
});