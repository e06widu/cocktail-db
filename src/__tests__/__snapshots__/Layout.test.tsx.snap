// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders Header component 1`] = `
<div>
  <header
    class="header "
    data-testid="header"
  >
    <div>
      <img
        alt="Logo"
        src="/logo.png"
      />
    </div>
    <div
      class="nav-items"
    >
      <div
        class="nav-item active"
      >
        Home
      </div>
      <div
        class="nav-item "
      >
        Search
      </div>
      <div
        class="nav-item "
      >
        Favorites 
      </div>
    </div>
  </header>
  <main
    class="main"
  >
    <div
      data-testid="child"
    >
      Child Component
    </div>
  </main>
</div>
`;
